### ISSUE TITLE
[Bug]: Sidebar scrollbar gets in the way of the expand/collapse button

### ISSUE THREAD

**Issue by kaiyoma** [storybookjs/storybook#24879]
### Describe the bug

The sidebar's scrollbar prevents clicking of the expand/collapse button.

### To Reproduce

https://github.com/storybookjs/storybook/assets/10719780/f96a9ba7-2560-4c7c-9302-a8da02e48f78

### System

```bash
Storybook Environment Info:

  System:
    OS: Windows 10 10.0.22621
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz
  Binaries:
    Node: 18.16.0 - C:\Program Files\Node.js\node.EXE
    npm: 9.8.0 - C:\Program Files\Node.js\npm.CMD
    pnpm: 8.6.6 - C:\Program Files\Node.js\pnpm.CMD <----- active
  Browsers:
    Edge: Chromium (119.0.2151.58)
  npmPackages:
    @storybook/addon-actions: 7.5.3 => 7.5.3
    @storybook/addon-controls: 7.5.3 => 7.5.3
    @storybook/react: 7.5.3 => 7.5.3
    @storybook/react-vite: 7.5.3 => 7.5.3
    storybook: 7.5.3 => 7.5.3
```

**Comment by suprabhat15** [comment]
@ndelangen Can i work on it?

**Comment by ndelangen** [comment]
What's the proposed solution?
@cdedreuille could you maybe get involved here?

**Comment by cdedreuille** [comment]
That's an issue I solve in 8.0. There's a `::before` background that help to extend the scrollbar area. On 7.x, as this is indeed a bug, we need to reduce the width of this background.

@suprabhat15 If you want to help fixing that, that would be really appreciated. I can review your PR and help if needed.

**Comment by suprabhat15** [comment]
Thanks @cdedreuille, for the context. Please assign it to me. I have some queries regarding the setup of the storybook. Can I ask here or in Discord?

**Comment by cdedreuille** [comment]
Thanks a lot @suprabhat15. I think Discord could be a good start. I'm sure someone from the community will be able to help you set up.

**Comment by ndelangen** [comment]
@suprabhat15 feel free to ping `@ndelangen` on discord (me)

**Comment by shilman** [comment]
For anyone who wants to try the fix in 8.0, try upgrading to the latest prerelease:

```
npx storybook@next upgrade --prerelease
```

**Comment by cdedreuille** [comment]
This is already fixed in Storybook 8.0. I'm closing this one for now.
